Doc/reference/compound_stmts.rst

```diff
@@ -550,6 +550,10 @@ is roughly equivalent to ::
 
 except that the original function is not temporarily bound to the name ``func``.
 
+.. versionchanged:: 3.9
+   Functions may be decorated with any valid expression.  Previously, the
```

Comment:
Maybe "... with any valid assignment expression". Ditto for the two other sentences you added.

Reply:
Sorry to be jumping back and forth between these two PRs.

I've changed the two notes in the Language Reference, since the linked `assignment_expression` nodes are well-defined in that context.

I'm resistant to changing the What's New wording, since the phrase "assignment expression" has a different meaning here... the walrus is just about the *last* thing that I want readers of this document to associate with the grammar change.

Using "expression" keeps with the wording in the PEP, which defines the term well.

Reply:
Good point. I will merge things tomorrow.